compiler/src/iree/compiler/Dialect/Util/IR/UtilOps.cpp

```diff
@@ -2159,6 +2191,21 @@ static void printListTypeSet(OpAsmPrinter &printer, Operation *, Type listType,
   }
 }
 
+LogicalResult ListConstructOp::verify() {
+  Operation *op = getOperation();
+  auto listType = cast<IREE::Util::ListType>(getResult().getType());
+  Type elementType = listType.getElementType();
+  for (auto value : llvm::enumerate(getValues())) {
+    Type valueType = value.value().getType();
+    if (!ListType::canImplicitlyCast(valueType, elementType)) {
+      return op->emitError()
+             << "list[" << value.index() << "] type " << valueType
+             << " cannot be be cast to list type " << elementType;
+    }
+  }
```

Comment:
nit: we can use structured binding.

```suggestion
  for (auto [idx, value] : llvm::enumerate(getValues())) {
    Type valueType = value.getType();
    if (!ListType::canImplicitlyCast(valueType, elementType)) {
      return op->emitError()
             << "list[" << idx << "] type " << valueType
             << " cannot be be cast to list type " << elementType;
    }
  }
```

Reply:
it's been so long since I wrote c++ :P thanks!